core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlRequestHandlerBase.java

```diff
@@ -298,22 +307,29 @@ private void setFinalResult(
       if (result.complete(resultSet)) {
         cancelScheduledTasks();
         throttler.signalSuccess(this);
-        long now = System.nanoTime();
-        long totalLatencyNanos = now - startTimeNanos;
-        long nodeLatencyNanos = now - callback.start;
-        context
-            .getRequestTracker()
-            .onNodeSuccess(statement, nodeLatencyNanos, executionProfile, callback.node);
-        context
-            .getRequestTracker()
-            .onSuccess(statement, totalLatencyNanos, executionProfile, callback.node);
-        session
-            .getMetricUpdater()
-            .updateTimer(
-                DefaultSessionMetric.CQL_REQUESTS,
-                executionProfile.getName(),
-                totalLatencyNanos,
-                TimeUnit.NANOSECONDS);
+
+        // Only call nanoTime() if we're actually going to use it
+        long completionTimeNanos = -1, totalLatencyNanos = -1;
+        if (!(requestTracker instanceof NoopRequestTracker)) {
+          completionTimeNanos = System.nanoTime();
```

Comment:
Contrary to what I suggested on the ticket, I decided not to reuse the time of the last `NodeResponseCallback.onResponse()`, because this method does a bit more stuff before updating the metrics (notably a call to `Conversions.toResultSet`).